lib/configure.js

```diff
@@ -334,6 +341,22 @@ module.exports = function() {
       if (err.message === 'repository not found') {
         process.stdout.write('I didn\'t find your repository in Travis.  Syncing Travis with GitHub…\n');
         return travis.users.sync.post()
+        .catch(function(err) {
+          // Ignore the exception if Travis is already syncing. The status code
+          // of the error is 409 and its message is 'Sync already in progress.
+          // Try again later.', but instead of relying on that we simply perform
+          // an explicit check.
+          return travisIsSyncing().then(function(is_syncing) {
+            if (!is_syncing) {
+              // Check again if it's active on Travis (in case it finished syncing
+              // right before we checked)
+              return ensureActiveInTravis().catch(function() {
+                // Throw the original error.
+                throw err;
+              });
+            }
+          });
+        })
         .then(travisAwaitSyncing)
```

Comment:
I've made travisAwaitSyncing unconditionally wait five seconds.
I haven't added the error message check, even if it would avoid some useless calls, because I couldn't assess if it was reliable enough (it isn't cited anywhere in the official documentation and it's never used in the official libraries for JavaScript and Ruby, moreover it isn't easily reproducible, so regressions would go unnoticed) and because this is such a rare case that a couple more calls shouldn't really matter.